File: java-breeds/app/src/main/java/es/tipolisto/breeds/data/model/RecordEntity.java
package es.tipolisto.breeds.data.model;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

//Esta anotación nos permite crear una entidad dentro de la base de datos
@Entity
public class RecordEntity {
    //Identificador único dentro de la base de datos
    @PrimaryKey(autoGenerate = true)
    private int id;
    @ColumnInfo(name="name")
    private String name;
    @ColumnInfo(name="date")
    private String date;
    @ColumnInfo(name="score")
    private int score;



    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }
}

File: kotlin-breeds/app/src/main/java/es/tipolisto/breeds/data/database/records/RecordEntity.kt
package es.tipolisto.breeds.data.database.records

import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.Entity


@Entity
data class RecordEntity (
    @PrimaryKey(autoGenerate = true) val id:Int?,
    @ColumnInfo(name = "name") val name: String,
    @ColumnInfo(name = "score") val score: Int,
    @ColumnInfo(name = "position") var position: Int,
    @ColumnInfo(name = "typeAnimal") var typeAnimal: String,
    @ColumnInfo(name = "date") val date: String
)
